library/src/main/java/com/otaliastudios/zoom/ZoomEngine.kt

```diff
@@ -638,42 +654,51 @@ internal constructor(context: Context) : ViewTreeObserver.OnGlobalLayoutListener
     @ScaledPan
     private fun checkPanBounds(horizontal: Boolean, allowOverScroll: Boolean): Float {
         @ScaledPan val value = if (horizontal) scaledPanX else scaledPanY
-        val viewSize = if (horizontal) mContainerWidth else mContainerHeight
+        val containerSize = if (horizontal) mContainerWidth else mContainerHeight
         @ScaledPan val contentSize = if (horizontal) mTransformedRect.width() else mTransformedRect.height()
-        @ScaledPan val basePanValue = computeBasePan()[(if (horizontal) 0 else 1)]
 
         val overScrollable = if (horizontal) mOverScrollHorizontal else mOverScrollVertical
         @ScaledPan val overScroll = (if (overScrollable && allowOverScroll) maxOverScroll else 0).toFloat()
-        return getPanCorrection(value, viewSize, contentSize, overScroll, basePanValue)
+        return getPanCorrection(value, containerSize, contentSize, overScroll, horizontal)
     }
 
     @ScaledPan
-    private fun getPanCorrection(@ScaledPan value: Float, viewSize: Float,
+    private fun getPanCorrection(@ScaledPan value: Float, containerSize: Float,
```

Comment:
while you're at it, could you add a documentation text to this method? It  is executed very frequently and takes a bunch of arguments that would benefit from additional documentation. Maybe the `checkPanBounds` method too.